### ISSUE TITLE
d.ts type imports are not resolved correctly in dependencies

### ISSUE THREAD

**Issue by dkjii-g** [microsoft/TypeScript#40581]
**TypeScript Version:**  4.1.0-dev.20200915

**Search Terms:**

path module resolution

**Code**

Please check this repo: https://github.com/dkjii-g/tsc-ref-bug

Check the README for bugs/instructions.

**Expected behavior:**

- The 4 folders compile correctly (they are slight variations to better explain the bug - you don't have to check them all).
- Typescript should not use the current project's module resolution when importing a different project's definitions.

**Actual behavior:**

Typescript tries to import repo2's modules from repo1's project.

This prevents the compiler from finding the type definition in dependencies.

**Related Issues:** #38111 #39117 (not sure)

**Comment by dkjii-g** [comment]
As a workaround, changing all the imports into relative paths makes it work.

**Comment by RyanCavanaugh** [comment]
A concise description of the problem may allow us to respond to this sooner, but generally these are misconfigurations or misunderstandings and will be looked at later.

**Comment by totony-misc** [comment]
Non-relative imports in dependent projects resolve according to the main project's baseUrl/tsconfig. That means that if a package uses non-relative imports and builds correctly (e.g. because resolution is done according to its own tsconfig), importing that package from another package will not work, since the generated index.js (entrypoint in its package.json) will have import directives which point to modules that are not resolvable from the importing packages' POV.

This might be a "work as intended," but it's really weird to have a package compile correctly, but being unable to import it in another because it uses non-relative imports.

**Comment by dkjii-g** [comment]
Non-relative imports in dependent projects resolve according to the main project's baseUrl/tsconfig. That means that if a package uses non-relative imports and builds correctly (because resolution is done according to its own tsconfig), importing that package from another package will not work, since the generated files of that package will have import directives which point to modules that are not resolvable from the importing package's POV.

This might be a "work as intended," but it's really weird to have a package compile correctly, but being unable to import it in another because it uses non-relative imports.

**Comment by rblaa** [comment]
I am running into this issue now myself.
Any progress or plans to fix?
